src/core/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -163,6 +164,13 @@ auto PenInputHandler::actionStart(InputEvent const& event) -> bool {
     // Forward event to page
     if (currentPage) {
         PositionInputData pos = this->getInputDataRelativeToCurrentPage(currentPage, event);
+        if (pos.pressure != Point::NO_PRESSURE) {
+            pressureMode = PressureMode::DEVICE_PRESSURE;
+        } else if (this->inputContext->getSettings()->isPressureGuessingEnabled()) {
```

Comment:
I think something like this could do the trick:
```suggestion
        } else if (this->inputContext->getSettings()->isPressureGuessingEnabled() || (stroke->size() > 0 && stroke->hasPressure())) {
```